.github/actions/ubuntu_18_setup/action.yml

```
@@ -48,7 +48,7 @@ runs:
         make prefix=/usr install
         git --version
 
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
```

```suggestion
    - uses: actions/checkout@v3
```
I think also this is not completely safe, rest looks good to me.

---

> I think also this is not right, and will fail in CI

I just reverted this change.